modules/tinymce/src/core/main/ts/fmt/FormatUtils.ts

```diff
@@ -169,6 +170,59 @@ const getParents = function (dom: DOMUtils, node: Node, selector?: string) {
   return dom.getParents(node, selector, dom.getRoot());
 };
 
+const isVariableFormatName = (editor: Editor, formatName: string): boolean => {
+  const hasVariableValues = (format: Format) => {
+    const isVariableValue = (val: string): boolean => val.length > 1 && val.charAt(0) === '%';
+    const stylesFieldHas = Obj.get(format, 'styles').exists((stylesField) => Arr.exists(Obj.values(stylesField), isVariableValue));
```

Comment:
This is somewhat inefficient, as you're looping over both attributes and styles, when you only need one to match. Additionally, attributes allows an array of names or a record of name/values (depends if it's a remove format).

As such you could rewrite this like so:
```
return Arr.exists([ 'styles', 'formats' ], (type) =>
  Obj.get(format, type).exists((field) => {
     const fieldValues = Type.isArray(field) ? field : Obj.values(field);
     return Arr.exists(field, isVariableValue);
  });
);
```